internal/app/prometheus-multi-tenant-proxy/reverse.go

```diff
@@ -46,16 +46,17 @@ func (r *ReversePrometheusRoundTripper) Director(req *http.Request) {
 func (r *ReversePrometheusRoundTripper) modifyRequest(req *http.Request, prometheusFormParameter string) error {
 
 	namespaces := req.Context().Value(Namespaces).([]string)
-	l := req.Context().Value(Labels).(map[string]string)
+	l := req.Context().Value(Labels).(map[string][]string)
 
 	// Convert the labels map into a slice of label matchers.
 	var labelMatchers []*labels.Matcher
 
 	for k, v := range l {
+		combinedValue := strings.Join(v, "|")
```

Comment:
I will add a new issue to this project.

I'm afraid of "promql" injection here.

Reply:
In addition, as per the code, `labelsvalues` would behave as an `or PromQL` selector. Am I Right?

Considering the target use case in Kubernetes environments, labels on resources work as an `and`... I'm unsure whether to ship this behavior.

On the other side, the behavior with namespaces is an or. So mixed feelings.

Let's ship it,